core/src/com/unciv/ui/worldscreen/unit/UnitTable.kt

```diff
@@ -163,9 +164,14 @@ class UnitTable(val worldScreen: WorldScreen) : Table(){
                     unitDescriptionTable.add(unit.promotions.XP.toString() + "/" + unit.promotions.xpForNextPromotion())
                 }
                 
-                if (unit.canSpreadReligion()) {
+                if (unit.canDoReligiousAction(Constants.spreadReligionAbilityCount)) {
                     unitDescriptionTable.add(ImageGetter.getStatIcon("Faith")).size(20f)
-                    unitDescriptionTable.add(unit.getReligionString())
+                    unitDescriptionTable.add(unit.getActionString(Constants.spreadReligionAbilityCount))
+                }
+                
+                if (unit.canDoReligiousAction(Constants.removeHeresyAbilityCount)) {
+                    unitDescriptionTable.add(ImageGetter.getImage("OtherIcons/Remove Heresy")).size(20f)
+                    unitDescriptionTable.add(unit.getActionString(Constants.removeHeresyAbilityCount))
```

Comment:
Hmmm this could also be unified but maybe we'll leave that until the third unit with X uses comes along.